build_tools/ci/cpu_comparison/run.py

```diff
@@ -37,18 +37,22 @@ def run_conv_test(config, filename, n_repeats):
     return True
 
 
-class RunOnTargetDevice(ABC):
+class BaseTemplate(ABC):
     """
     Base class to be inherited by any new dispatch/compute op template.
     The derived instances would therefore be created specifying the intended target
     device(s) they're to be run; and would accordingly be `run` if the intended
     target device(s) contains the `target_device` supplied via command-line.
     """
 
-    def __init__(self, run_on_target):
+    def __init__(self, run_on_target, additional_aie_compilation_flags=None):
         self.run_on_target = run_on_target
+        self.additional_aie_compilation_flags = additional_aie_compilation_flags \
+            if additional_aie_compilation_flags is not None else []
```

Comment:
Can you just do `__init__(self, run_on_target, additional_aie_compilation_flags=[])`
and `self.additional_aie_compilation_flags = additional_aie_compilation_flags`?

Reply:
I prefer not doing that because it can lead to potential issues (although likely not in this case): https://stackoverflow.com/questions/43255481/python-default-list-in-function

Reply:
Just noticed: you are overriding the additional flags for every test which runs subsequent to the one that appends. I'll try and think of a way to add flags to a test in isolation